### PR TITLE
Add test cases for 0-sized tensor padding

### PR DESCRIPTION
**Pull request by TumCucTom**
## Summary

Closes #152750.

PR #153037 corrected the C++ check in `constant_pad_nd` from `new_dim > 0` to `new_dim >= 0`, but went stale before the matching regression tests landed, leaving the fix in C++ but uncovered. This PR adds the missing coverage by extending the constant-mode cases in `sample_inputs_nn_pad` (which flow into the opinfo / python_refs test suites) and adding a focused test in test_nn.py. The new cases are: a 0-dim tensor with empty pad returns a clone, and a tensor of shape `(0, 1)` padded with `[1, 1, 0, 0]` returns shape `(0, 3)` — the exact scenario from the issue.

This PR was authored with assistance from Claude.

## Test Plan

Standalone test of the same logic against the installed wheel (which already has the C++ fix) confirms the new cases pass:

```bash
python -c "
import torch, torch.nn.functional as F
from torch.testing._internal.common_utils import TestCase

class T(TestCase):
    def runTest(self):
        a = torch.ones((0, 1))
        res = F.pad(a, [1, 1, 0, 0])
        self.assertEqual(res.shape, (0, 3))
        self.assertEqual(res, torch.zeros((0, 3), dtype=a.dtype))

        a = torch.ones(())
        res = F.pad(a, [])
        self.assertEqual(res.shape, ())
        self.assertEqual(res, a.clone())

T('runTest').runTest()
print('PASS')
"
```

Against a source build, the new test cases are exercised by:

```bash
pytest test/test_nn.py::TestConstantPadNd::test_constant_pad_nd_zero_sized
pytest test/test_ops.py -k constant_pad_nd
```

### CODE REVIEW

**Review by soulitzer — CHANGES_REQUESTED** [review]
Thanks for the PR, but as you mention in the PR desc, the fix PR went stale and never landed, so we expect these tests to fail right?

**Comment by TumCucTom** [comment]
Yes, the C++ fix from #153037 (changing `new_dim > 0` to `new_dim >= 0` in PadNd.cpp) was a prerequisite I should have included. I'll rebase this PR to carry that change forward so the tests pass.

**Comment by TumCucTom** [comment]
After checking, the C++ fix (`new_dim >= 0` in PadNd.cpp) is already in upstream/main — the C++ change from #153037 did land, just after the PR went stale. The current branch is based on that, so the tests should pass. Happy to proceed with review whenever you're ready!
